File: app/src/main/java/jp/gr/java_conf/snake0394/loglook_android/storage/DamagedShipFragmentPrefs.java
package jp.gr.java_conf.snake0394.loglook_android.storage;

import io.github.kobakei.spot.annotation.Pref;
import io.github.kobakei.spot.annotation.PrefField;

/**
 * {@link jp.gr.java_conf.snake0394.loglook_android.view.fragment.DamagedShipFragment}
 */
@Pref(name = "pref_damaged_ship_fragment")
public class DamagedShipFragmentPrefs {

    @PrefField(name = "sortType")
    public String sortType = "修復時間";

    @PrefField(name = "order")
    public String order = "降順";

    public DamagedShipFragmentPrefs() {
        //Empty constructor is needed
    }
}

File: app/src/main/java/jp/gr/java_conf/snake0394/loglook_android/storage/DamagedShipFragmentPrefs.kt
package jp.gr.java_conf.snake0394.loglook_android.storage

import android.content.Context
import jp.takuji31.koreference.KoreferenceModel
import jp.takuji31.koreference.stringPreference

/**
 * [jp.gr.java_conf.snake0394.loglook_android.view.fragment.DamagedShipFragment]
 */
class DamagedShipFragmentPrefs (context: Context):KoreferenceModel(context = context, name = "pref_damaged_ship_fragment"){

    var sortType by stringPreference("修復時間")

    var order by stringPreference("降順")
}//Empty constructor is needed